includes/admin/class-wc-stripe-settings-controller.php

```diff
@@ -272,6 +272,11 @@ public function admin_scripts( $hook_suffix ) {
 	 * to display the payment gateways on the WooCommerce Settings page.
 	 */
 	public static function hide_gateways_on_settings_page() {
+		// Prevent hiding gateways in the new payments settings experience (React-based UI).
+		if ( 'yes' === get_option( 'woocommerce_feature_reactify-classic-payments-settings_enabled' ) ) {
```

Comment:
Side note: it feels like we shouldn't need to depend on the raw option name to work out whether this feature is enabled or not. I'll kick off an internal thread to see whether a wrapper is reasonable, and I may well move to an issue in WooCommerce.

Reply:
It looks like we may be able to do this in a way that includes default values etc:
```suggestion
		$woo_feature_controller = wc_get_container()->get( \Automattic\WooCommerce\Internal\Features\FeaturesController::class );
		if ( $woo_feature_controller && $woo_feature_controller->feature_is_enabled( 'reactify-classic-payments-settings' ) ) {
```

Reply:
@daledupreez Thanks for pointing me in the right direction!

Looks like there's a [utility class](https://github.com/woocommerce/woocommerce/blob/02669dcfc2e7a379ac8795966815a4e36d9cd30a/plugins/woocommerce/src/Utilities/FeaturesUtil.php#L41) we can use, I've updated it in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/4294/commits/2ee1c4d0b4890efdab58b9c57c4b530a316ad90e